### DISCUSSION TITLE
沉浸式翻译链接azure OpenAI的问题

### DISCUSSION THREAD

**Discussion by joeaki1983** [immersive-translate/immersive-translate#1160] [Help] [answered]
### 概述 | Summary

在沉浸式翻译里用azure的OpenAI API，按照说明文档设置的，提示服务返回错误：400， 说明请求参数错误，请检查相应配置。 Unsupported data type  ，已经检查过配置无误，该如何解决，谢谢。

**Comment by theowenyoung** [comment]
请截图一下你的沉浸式翻译 openai 设置

joeaki1983 ***@***.***> 于 2023年11月30日周四 上午12:53写道：

**Comment by joeaki1983** [comment]
![image](https://github.com/immersive-translate/immersive-translate/assets/10794638/7818d57c-0610-43c7-bbcb-25f4c401e9ae)
![image](https://github.com/immersive-translate/immersive-translate/assets/10794638/a78d7e91-c8f6-447c-b50e-363fd83e1d42)

确认api等信息输入无误。

**Reply by theowenyoung** [reply]
看起来是没有什么问题的，你再确实下地址？ 在playground里面选择 completion，然后复制 endpoint，看看有没有不一样：

![CleanShot 2023-11-30 at 01 35 00@2x](https://github.com/immersive-translate/immersive-translate/assets/62473795/39eac151-22d8-4471-9c4d-533e77f76a74)


这是我的设置，测试是可以成功的：

![CleanShot 2023-11-30 at 01 37 07@2x](https://github.com/immersive-translate/immersive-translate/assets/62473795/960f7f3c-70fe-4f94-bad3-bed9e06f18ca)

**Reply by joeaki1983** [reply]
那个链接就是整个从那里面复制出来的，是不是和建立的模型有关系，我测试的是gpt3.5-0301的模型，可是0613的也试过了，也是不行。azure 的API确定是好的，因为在chatbox里可以正常使用。

**Reply by theowenyoung** [reply, marked as answer]
https://stackoverflow.com/questions/75820041/unsupported-data-type-when-making-api-request-to-azure-open-ai

搜了一下，你好像是 缺了路径里的 chat

**Reply by joeaki1983** [reply]
囧，多谢大佬，果然是这样，已经可以用了，半夜还回答问题，真敬业，感谢！
